modules/node/init.zsh

```diff
@@ -6,13 +6,43 @@
 #   Zeh Rizzatti <zehrizzatti@gmail.com>
 #
 
+lazy_load_nvm() {
+  NVM_DIR=$1
+  # Skip adding binaries if there is no node version installed yet
+  if [ -d $NVM_DIR/versions/node ]; then
+    NODE_GLOBALS=(`find $NVM_DIR/versions/node -maxdepth 3 \( -type l -o -type f \) -wholename '*/bin/*' | xargs -n1 basename | sort | uniq`)
+  fi
+  NODE_GLOBALS+=("nvm")
+
+  load_nvm () {
+    # Unset placeholder functions
+    for cmd in "${NODE_GLOBALS[@]}"; do unset -f ${cmd} &>/dev/null; done
+
+    # Load NVM
+    [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
+
+    # (Optional) Set the version of node to use from ~/.nvmrc if available
+    nvm use 2> /dev/null 1>&2 || true
+
+    # Do not reload nvm again
+    export NVM_LOADED=1
+  }
+
+  for cmd in "${NODE_GLOBALS[@]}"; do
+    # Skip defining the function if the binary is already in the PATH
+    if ! which ${cmd} &>/dev/null; then
+      eval "${cmd}() { unset -f ${cmd} &>/dev/null; [ -z \${NVM_LOADED+x} ] && load_nvm; ${cmd} \$@; }"
```

Comment:
Unsetting this specific command and using NVM_LOADED shouldn't be needed because we unset all the functions in load_nvm anyway, so we don't need to track it.